src/main/scala/com/advancedtelematic/director/db/EcuUpdateAssignmentMigration.scala

```diff
@@ -0,0 +1,60 @@
+package com.advancedtelematic.director.db
+
+import akka.actor.ActorSystem
+import akka.stream.Materializer
+import akka.stream.scaladsl.{Sink, Source}
+import akka.Done
+import com.advancedtelematic.libats.data.DataType.Namespace
+import com.advancedtelematic.libats.slick.db.SlickValidatedGeneric.validatedStringMapper
+import org.slf4j.LoggerFactory
+import slick.jdbc.GetResult
+import slick.jdbc.MySQLProfile.api._
+import com.advancedtelematic.libats.slick.db.SlickAnyVal._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+
+class EcuUpdateAssignmentMigration(implicit
+  val db: Database,
+  val mat: Materializer,
+  val system: ActorSystem,
+  val ec: ExecutionContext)
+  extends DeviceUpdateAssignmentRepositorySupport
+  with EcuUpdateAssignmentRepositorySupport {
+
+  private val _log = LoggerFactory.getLogger(this.getClass)
+
+  implicit val getResult: GetResult[Namespace] = pr => {
+    implicitly[ColumnType[Namespace]].getValue(pr.rs, 1)
+  }
+
+  def run: Future[Done] = {
+    val source = db.stream(Schema.deviceTargets.result)
+    Source.fromPublisher(source).mapAsyncUnordered(1) { case oldAssignment =>
+      val queryNamespace = sql"""SELECT DISTINCT(namespace) from `ecus` WHERE device = '#${oldAssignment.device}'""".as[Namespace]
+      db.run(queryNamespace).flatMap { namespace =>
```

Comment:
Isn't there a way to do this with a single `db.run`? Maybe something similar to this (I haven't tested it)?:
```
val source = db.stream(Schema.deviceTargets.result)
    Source.fromPublisher(source).mapAsyncUnordered(1) { oldAssignment =>
      val queryNamespaces = sql"""SELECT DISTINCT(namespace) from `ecus` WHERE device = '#${oldAssignment.device}'""".as[Namespace]
      db.run {
        for {
          namespaces <- queryNamespaces
          assignment <- deviceUpdateAssignmentRepository.persistAction(
            namespaces.head,
            oldAssignment.device,
            oldAssignment.correlationId,
            oldAssignment.updateId,
            oldAssignment.targetVersion)
          targets <- ecuUpdateAssignmentRepository.legacyFetchAction(
            assignment.namespace,
            assignment.deviceId,
            assignment.version)
          _ <- ecuUpdateAssignmentRepository.persistAction(
            assignment.namespace,
            assignment.deviceId,
            assignment.version,
            targets)
        } yield ()
      }
    }.runWith(Sink.foreach { _ => () })
```